text/0000-Std-Set-Impl.md

```diff
@@ -0,0 +1,264 @@
+```
+let S = { 1, 2, 3, 6 }
+let T = { 1, 2, 3, 4, 5 }
+```
+The `union` of `S` and `T` is as follows:
+```
+V = { 1, 2, 3, 4, 5, 6 }
+```
+
+Intersections:
+Given two sets, the `intersection` of the sets is a set which contains all elements which are elements of both sets.
+
+```
+let S = { 1, 2, 3, 4, 5 }
+let T = { 3, 4, 5, 6, 7 }
+```
+The `intersection` of `S` and `T` is as follows:
+```
+V = { 3, 4, 5 }
+```
+In the case that two sets share no common elements, then the intersection of the two sets is the empty set, and the sets are said to be `disjoint`.
+
+With these formal definitions, we have sufficient information to begin describing the implementation.
+
+## Data Structures
+
+Set:
+```pony
+  class Set
+    """
+     Docs
+    """
+    create // todo
+```
+
+Empty Set:
+```
+  static Set Empty = Set{ } 
+```
+## Methods
+
+Equality:
+```
+  Bool is_equal(set: Set)
+```
+
+Cardinality:
+```
+  uint cardinality()
+```
+
+Equivalence:
+```
+  Bool is_equivalent(set: Set)
+```
+
+Subsets:
+```
+  Bool is_subset(set: Set)
+  Bool is_proper_subset(set: Set)
+```
+
+Supersets: 
+```
+  Bool is_superset(set: Set)
+  Bool is_proper_superset(set: Set)
+```
+
+Empty:
+```
+  Bool is_empty_set()
+```
+
+Complementary:
+```
+  Set get_complementary_set(set: Set)
+  Bool is_complementary_set(set: Set, complement: Set)
+```
+
+Unions:
+```
+  Set get_union(set: Set)
+  Bool is_union(set: Set, union: Set)
+```
+
+Intersections:
+```
+  Set get_intersection(set: Set)
+  Bool is_intersection(set: Set, intersect: Set)
+  Bool are_disjoint(set: Set)
+```
+
+Validity:
+```
+  Bool is_valid_set()
+```
+
+Operators:
+```
+  Set A == Set B => A.is_equal(B)
+  Set A != Set B => !A.is_equal(B)
+  Set A >= Set B => A.is_superset(B)
+  Set A >  Set B => A.is_proper_superset(B)
+  Set A <= Set B => A.is_subset(B)
+  Set A <  Set B => A.is_proper_subset(B)
+  Set A |  Set B => A.get_union(B)
+  Set A &  Set B => A.get_intersection(B)
+```
+
+
+# How We Teach This
+
+The best terminology for this library would be to stick with pure set theory terminology. There likely won't be much to teach, as anyone who will be using this library will have some familiarity with set theory, and it's terminology. Set Theorists who will be the target consumers of this library will feel most comfortable with their usual terms.
+
+# How We Test This
+ Unit tests will be created.
+
+# Drawbacks
+Adding this to the standard library will have some maintainence cost associated. 
+
+# Alternatives
+Some alternatives could be other naming conventions. Some Set Theory terminology can seem quite nuanced to novices, especially with terms like "Equal" and "Equivalent" having very different meanings. Another alternative would be completely leaving this structures and methods out of the standard library and using it as a free-standing thrid-party library.
+
+# Unresolved questions
+What is the most idiomatic way to structure this class?
+Can operators be overloaded?
```

Comment:
The existing operators can be overloaded. See https://tutorial.ponylang.io/expressions/infix-ops.html